src/api/app/controllers/staging/staged_projects_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Staging::StagedProjectsController < ApplicationController
```

Comment:
Why StagedProjects? We already have app/controllers/staging/staging_projects_controller.rb ?

Reply:
the issue here is the routing. As the routes are today, it wouldn't work like that. It's not under ```StagingWorkflow``` and therefore, the route wouldn't accept the project_name as parameter. The route would be /staging/staging_projects. /staging/staging_projects would lead to show. If you look in the webui, we scoped everything under 'staging_workflow'.

Reply:
Most clever would be to use one controller. But that already didn't happen for the other parts...

Reply:
The problem is that you miss the nested resource, you basically have two projects: main project and the staging project. You need to put it into a nested resource

``
resources :staging, only: [], param: 'main_project_name' do
 resources :staging_projects, only: [:index, :show], param: :name do
 end
end
``

It would produce routes like:

```
/staging/:staging_main_project_name/staging_projects
/staging/:staging_main_project_name/staging_projects/:staging_project_name
```

This is how rails suggests it
https://guides.rubyonrails.org/routing.html#nested-resources

Of course this requires that you always add the main project to the route but I don't think this is an issue and also Rails does it in their official documentation.

Reply:
> Most clever would be to use one controller. But that already didn't happen for the other parts...

now its the time to change, after will be hard. The "other parts" came before everything ;)

Reply:
> now its the time to change

Well, depends. We are going to iterate over this for sure. So if can also be something for the next deliverable.